transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpSubjectSecurityFilter.java

```diff
@@ -357,32 +356,17 @@ void securityMessageReceived(NextFilter nextFilter, IoSession session, Object me
 
         DefaultAuthenticationToken authToken = (DefaultAuthenticationToken) tokenExtractor.extract(httpRequest, realm);
 
-        // If the client request provided authentication data which has
```

Comment:
You have correctly changed this logic to be a 401 instead of 400. 
AND you have moved this logic into the the HttpLoginSecurityFilter.login().  

The HttpLoginSecurityFilter.login() is in a scheduled task with suspends incoming data.  Yet, the logic you are checking does not require suspended reads nor a scheduled task.  So it would be better if the logic you implemented remained in this thread, rather than being scheduled.  

If there are practical reasons why it needs to be in that method that I'm not aware of, please mention them